src/lsp/cobol_parser/grammar.mly

```diff
@@ -623,18 +623,76 @@ let environment_division :=
 
 (* ------------- ENVIRONMENT DIVISION / CONFIGURATION SECTION -------------- *)
 
-let configuration_section :=
- | CONFIGURATION; SECTION; ".";
-   source_computer_paragraph = ro(loc(source_computer_paragraph));
-   object_computer_paragraph = ro(loc(object_computer_paragraph));
-   special_names_paragraph   = ro(loc(special_names_paragraph));
-   repository_paragraph      = ro(loc(repository_paragraph));     (* +COB2002 *)
-   { { source_computer_paragraph;
-       object_computer_paragraph;
-       special_names_paragraph;
-       repository_paragraph } }
+(*
+(* order *)
+let unorder2_opt (A, B) ==
+ | a = A; b = ro(B); { (P1 (Some a), P2 b) }
+ | b = B; a = ro(A); { (P2 a, P1 (Some b)) }
+ | { (P1 None, P2 None) }
+let unorder3_opt (A, B, C) ==
+ | a = A; u = unorder2_opt(B, C); { let (b, c) = u in (P1 (Some a), ~+!b, ~+!c) }
+ | b = B; u = unorder2_opt(A, C); { let (a, c) = u in (~+!a, P1 (Some b), ~+!c) }
+ | c = C; u = unorder2_opt(A, B); { let (a, b) = u in (~+!a, ~+!b, P1 (Some c)) }
+ | { (P1 None, P2 None, P3 None) }
+let unorder4_opt (A, B, C, D) ==
+ | a = A; u = unorder3_opt(B, C, D); { let (b, c, d) = u in (P1 (Some a), ~+!b, ~+!c, ~+!d) }
+ | b = B; u = unorder3_opt(A, C, D); { let (a, c, d) = u in (~+!a, P1 (Some b), ~+!c, ~+!d) }
+ | c = C; u = unorder3_opt(A, B, D); { let (a, b, d) = u in (~+!a, ~+!b, P1 (Some c), ~+!d) }
+ | d = D; u = unorder3_opt(A, B, C); { let (a, b, c) = u in (~+!a, ~+!b, ~+!c, P1 (Some d)) }
+ | { (P1 None, P2 None, P3 None, P4 None) }
+ *)
 
+(*
+(* whole  *)
+let unorder2_opt (A, B) ==
+ | a = A; b = ro(B); { (Some a, b, AB) }
+ | b = B; a = ro(A); { (a, Some b, BA) }
+ | { (None, None, AB) }
+let unorder3_opt (A, B, C) ==
+ | a = A; u = unorder2_opt(B, C); { let (b, c, o) = u in (Some a, b, c, from_duo `BC o) }
+ | b = B; u = unorder2_opt(A, C); { let (a, c, o) = u in (a, Some b, c, from_duo `AC o) }
+ | c = C; u = unorder2_opt(A, B); { let (a, b, o) = u in (a, b, Some c, from_duo `AB o) }
+ | { (None, None, None, ABC) }
+let unorder4_opt (A, B, C, D) ==
+ | a = A; u = unorder3_opt(B, C, D); { let (b, c, d, o) = u in (Some a, b, c, d, from_trio `BCD o) }
+ | b = B; u = unorder3_opt(A, C, D); { let (a, c, d, o) = u in (a, Some b, c, d, from_trio `ACD o) }
+ | c = C; u = unorder3_opt(A, B, D); { let (a, b, d, o) = u in (a, b, Some c, d, from_trio `ABD o) }
+ | d = D; u = unorder3_opt(A, B, C); { let (a, b, c, o) = u in (a, b, c, Some d, from_trio `ABC o) }
+ | { (None, None, None, None, ABCD) }
+ *)
+
+let unorder2_opt (A, B) ==
+ | a = A; b = ro(B); { (Some a, b, AB) }
+ | b = B; a = ro(A); { (a, Some b, BA) }
+ | { (None, None, AB) }
+let unorder3_opt (A, B, C) ==
+ | a = A; u = unorder2_opt(B, C); { let (b, c, o) = u in (Some a, b, c, from_duo' `BC o) }
```

Comment:
Couldn't we have something that reads `prepend_as_trio `A `BC o` directly instead?
Or even something parametric like:
```suggestion
 | a = A; u = unorder2_opt(B, C); { prepend_as_trio `A u }
```
with the implicit that `u` encodes a duo `BC` thanks to the ``A`?

Reply:
Also I think for the sake of the automaton size, I'd be in favor to try using non-inlined items.